src/operation_dispatcher.rs

```diff
@@ -162,14 +195,14 @@ impl OperationDispatcher {
                                         // We index only if it was just transitioned to Waiting after triggering
                                         self.waiting_operations.insert(token_id, operation.clone());
                                         // TODO(didierofrivia): Decide on indexing the failed operations.
-                                        Some(operation.clone())
+                                        Ok(operation.clone())
                                     }
                                     State::Done => self.next(),
                                 }
                             }
-                            Err(status) => {
-                                error!("{status:?}");
-                                None
+                            Err(err) => {
+                                error!("{err:?}");
```

Comment:
I wonder if we need to log here? error is being returned

Reply:
we could, we were using it for debugging. not strong opinion about keeping it

Reply:
let me know and I can remove it

Reply:
leave it, not a big issue.

Reply:
Right, this is... a little random. Logging at _error creation_ time, possibly at a _lower_ level, e.g. `debug`, is something I'd get. Here,... what's the point? In the middle of the call stack we grab the error and log it 🤔 

Might be because you had to deal with it to "only" rewrap it? Cause you could actually get away with _not_ doing that with the [`?`](https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html)